packages/file-entry-cache/test/index.test.ts

```diff
@@ -160,6 +161,36 @@ describe('getFileDescriptor()', () => {
 		expect(fileDescriptor.meta.data).to.not.toBeDefined();
 	});
 
+	test('should save the meta data after the first call and loading data', () => {
+		const shared = {shared: 'shared'};
+		const data = {testingFooVariable: '11', name: 'test1.txt', shared};
+		const fileEntryCache = new FileEntryCache({useCheckSum: true});
+		const testFile1 = path.resolve('./.cacheGFD/test1.txt');
+		const fileDescriptor = fileEntryCache.getFileDescriptor(testFile1);
+		fileDescriptor.meta.data = data;
+		expect(fileDescriptor).toBeDefined();
+		fileEntryCache.reconcile();
+
+		// Add the meta data to the cache
+		const fileEntryCache2 = createFromFile(fileEntryCache.cache.cacheFilePath, true);
+		const fileDescriptor2 = fileEntryCache2.getFileDescriptor(testFile1);
+		const data2 = {testingFooVariable: '22', name: 'test1.txt', shared};
+		fileDescriptor2.meta.data = data2;
+		fileEntryCache2.reconcile();
+
+		// Load the meta data from the cache
+		const fileEntryCache3 = createFromFile(fileEntryCache.cache.cacheFilePath, true);
+		const fileDescriptor3 = fileEntryCache3.getFileDescriptor(testFile1);
+		expect(fileDescriptor3).toBeDefined();
+		expect(fileDescriptor3.meta.data).toEqual(data2);
+
+		// Verify that the data shows changed
+		const fileDescriptor4 = fileEntryCache3.getFileDescriptor(testFile1);
+		expect(fileDescriptor4).toBeDefined();
+		expect(fileDescriptor4.meta.data).toEqual(data2);
+		expect(fileDescriptor4.changed).toEqual(true);
```

Comment:
Why do you expect `changed` to be true?

The file didn't change. Yes, `meta.data` had gotten added.

In my case, I only care if the file changed, not the `meta.data`. I'm ok with being responsible for detecting changes to the `meta.data` if necessary.